### ISSUE TITLE
Bug in alias annotations

### ISSUE THREAD

**Issue by hjalmarlucius** [microsoft/pyright#7184]
# Environment data

* pyright 1.1.349
* python version 3.11.6

# Code snippet

```py
from typing import Generic
from typing import Literal
from typing import TypeAlias
from typing import TypeVar

A_float: TypeAlias = Literal["float32", "float64"]
A_numeric: TypeAlias = Literal["int32", "int64"] | A_float
Tdtype = TypeVar("Tdtype", bound=A_numeric)


class TypedArray(Generic[Tdtype]):
    pass


Tdtype_float = TypeVar("Tdtype_float", bound=A_float)

# WORKS
OptionalFloatArray: TypeAlias = TypedArray[Tdtype_float] | None

# FAILS but equivalent
OptionalArray: TypeAlias = TypedArray[Tdtype] | None
OptionalFloatArray_: TypeAlias = OptionalArray[Tdtype_float]
# error: Could not specialize type "OptionalArray[Tdtype@OptionalArray]"
#   "Literal['float32']" cannot be assigned to type "Literal['int32']"
#   "Literal['float64']" cannot be assigned to type "Literal['int64']"
```

# Error description

Valid aliases with TypeVars raise error if Literal bounds are narrowed.
* Seems to only happen when using Literals in both bounds
* Does not happen if there is only one Literal in `A_float`
* Error description indicates that the bug is caused by doing subtype comparison in the wrong order

**Comment by erictraut** [comment]
Thanks for the bug report. I'm able to repro, and it does appear to be a bug.

Here's a simplified repro, for my own reference when investigating further.

```python
from typing import Literal

type TA1[T2: Literal["b", "c"]] = list[T2]

type TA2[T1: Literal["a", "b", "c"]] = list[T1]
type TA3[T2: Literal["b", "c"]] = TA2[T2]
```

**Comment by erictraut** [comment]
This will be addressed in the next release of pyright.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.350, which I just published.
